File: extract_pdfs.py
import os
import PyPDF2

docs_dir = 'docs'
for filename in os.listdir(docs_dir):
    if filename.endswith('.pdf'):
        filepath = os.path.join(docs_dir, filename)
        outpath = filepath + '.txt'
        print(f'Extracting {filepath}...')
        try:
            with open(filepath, 'rb') as f:
                reader = PyPDF2.PdfReader(f)
                text = []
                for page in reader.pages:
                    text.append(page.extract_text() or '')
                
            with open(outpath, 'w', encoding='utf-8') as f:
                f.write('\n'.join(text))
            print(f'Done: {outpath}')
        except Exception as e:
            print(f'Error extracting {filepath}: {e}')

File: extract_pdfs.mjs
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import pdfParse from 'pdf-parse';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function extractText(filePath) {
    try {
        console.log(`Extracting text from ${filePath}...`);
        const dataBuffer = fs.readFileSync(filePath);
        const data = await pdfParse(dataBuffer);
        fs.writeFileSync(filePath + '.txt', data.text);
        console.log(`Successfully extracted text to ${filePath}.txt`);
    } catch (err) {
        console.error(`Error processing ${filePath}:`, err);
    }
}

async function main() {
    const docsDir = path.join(__dirname, 'docs');
    const files = fs.readdirSync(docsDir);
    for (const file of files) {
        if (file.toLowerCase().endsWith('.pdf')) {
            await extractText(path.join(docsDir, file));
        }
    }
}

main();
